tests/integration/helm/util.go

```diff
@@ -156,20 +166,47 @@ func deleteIstio(t framework.TestContext, h *helm.Helm, cs *kube.Cluster) {
 }
 
 // VerifyInstallation verify that the Helm installation is successful
-func VerifyInstallation(ctx framework.TestContext, cs cluster.Cluster) {
+func VerifyInstallation(ctx framework.TestContext, cs cluster.Cluster, verifyGateways bool) {
 	scopes.Framework.Infof("=== verifying istio installation === ")
 
 	retry.UntilSuccessOrFail(ctx, func() error {
 		if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(cs, IstioNamespace, "app=istiod")); err != nil {
 			return fmt.Errorf("istiod pod is not ready: %v", err)
 		}
-		if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(cs, IstioNamespace, "app=istio-ingressgateway")); err != nil {
-			return fmt.Errorf("istio ingress gateway pod is not ready: %v", err)
-		}
-		if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(cs, IstioNamespace, "app=istio-egressgateway")); err != nil {
-			return fmt.Errorf("istio egress gateway pod is not ready: %v", err)
+
+		if verifyGateways {
+			if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(cs, IstioNamespace, "app=istio-ingressgateway")); err != nil {
+				return fmt.Errorf("istio ingress gateway pod is not ready: %v", err)
+			}
+			if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(cs, IstioNamespace, "app=istio-egressgateway")); err != nil {
+				return fmt.Errorf("istio egress gateway pod is not ready: %v", err)
+			}
 		}
 		return nil
 	}, retry.Timeout(RetryTimeOut), retry.Delay(RetryDelay))
 	scopes.Framework.Infof("=== succeeded ===")
 }
+
+func SetRevisionTag(ctx framework.TestContext, h *helm.Helm, fileSuffix, revision, revisionTag string) {
```

Comment:
So will we be documenting to template from the revision tag chart and the `kubectl apply` to Helm users? I think when switching tags this is the only way that works(?) since otherwise there'd be chart overlap conflicts

Maybe in the future we have a dedicated chart for tags or just put it all in base so users could `helm upgrade` instead.

Reply:
For now yes. Users are also able to specify using `istioctl`, but for those that cannot this is a current workaround. When you template it only does it for a specific template that you provide. This way its declarative, but isn't kept to a helm chart.

We'd have to think through how users should use helm install/upgrade, but @nrjpoddar and I were ok with this approach. We can always re-evaluate it after we get feedback from users. For now, this is about creating parity for Helm users with `istioctl` users.

Reply:
While this is a workaround to get parity, if everyone is ok with restructuring Helm chart or adding/moving templates around I’m up for that. @Monkeyanator i don’t quite get how installing a new mutating webhook can be achieved though.

Only way I can think of is having 2 copies of mutating webhook where the second one is guarded by a Helm value “defaultRevision” being set? We can do that now and then just the Helm upgrade workflow would work.

Reply:
After talking about this with @Monkeyanator we agreed to write up an RFC for addressing the chart structure. The current base chart has deprecated items and in https://github.com/istio/istio/pull/33773 the webhooks are being moved from discovery to base.

Being consistent would be helpful for devs and users to get the experience they desire. Helm will probably go to Beta in 1.11, but without a clear plan it'll be harder to reach the Stable stage.